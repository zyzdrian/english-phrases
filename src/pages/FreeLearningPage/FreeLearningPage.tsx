import React from 'react';
import { RouteComponentProps } from '@reach/router';
import styled from 'styled-components';
import usePhrases from '../../hooks/usePhrases';
import { TopBar } from '../../components';
import { Phrase } from '../../components';

export enum FREE_LEARNING_PAGE_TEST_IDS {
    phrasesContainer = 'phrases-container'
}

const StyledFreeLearningPage = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  min-height: 100vh;
  overflow: hidden;
`;

const StyledContainer = styled.div`
display: grid;
grid-template-rows: 240px;
grid-template-columns: 552px 552px;
grid-gap: 16px;

@media screen and (max-width: 1200px) {
  grid-template-columns: 552px;
}
@media screen and (max-width: 600px) {
  grid-template-columns: calc(100vw - 48px);
}
`;

export const FreeLearningPage: React.FC<RouteComponentProps> = () => {
  const { phrases } = usePhrases();
  return (
    <StyledFreeLearningPage>
      <TopBar title="Free Learning" />
      <StyledContainer data-testid={FREE_LEARNING_PAGE_TEST_IDS.phrasesContainer}>
        {phrases.map((phrase) => (
          <Phrase key={phrase.id} phrase={phrase} />
        ))}
      </StyledContainer>
    </StyledFreeLearningPage>
  );
};
