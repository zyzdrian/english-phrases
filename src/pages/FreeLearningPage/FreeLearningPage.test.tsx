import React from 'react';
import { render } from '@testing-library/react';
import { FreeLearningPage, FREE_LEARNING_PAGE_TEST_IDS } from './FreeLearningPage';
import '@testing-library/jest-dom/extend-expect';

jest.mock('../../hooks/usePhrases', () => () => ({
    phrases: [
        {
            id: '1',
            name: 'Test phrase',
            appropriate: 'Test appropriate',
            chapterId: 1,
            meanings: ['Meaning'],
            bucketIndex: 1
        },
        {
            id: '2',
            name: 'Test phrase 2',
            appropriate: 'Test appropriate',
            chapterId: 1,
            meanings: ['Meaning 1', 'Meaning 2'],
            bucketIndex: 1
        }
    ]
    })
);


describe('<FreeLearningPage />', () => {
    it('should return phrases page with phrases', () => {
        const phrasesLength = 2;
        const { container, getByTestId } = render(<FreeLearningPage />);

        expect(container.firstChild).toMatchSnapshot();
        expect(getByTestId(FREE_LEARNING_PAGE_TEST_IDS.phrasesContainer).children.length).toEqual(phrasesLength);
    });
});
