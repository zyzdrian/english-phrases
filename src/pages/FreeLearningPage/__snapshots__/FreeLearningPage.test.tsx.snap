// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FreeLearningPage /> should return phrases page with phrases 1`] = `
<div
  class="sc-kAzzGY cYQrAh"
>
  <div
    class="sc-VigVT cybuwg"
  >
    <img
      alt="menu"
      class="sc-jTzLTM kMqZix"
      data-testid="hamburger"
      src="menu.svg"
    />
    <div
      class="sc-fjdhpX jmFvEE"
      data-testid="title"
    >
      Free Learning
    </div>
    <span
      style="width: 24px;"
    />
  </div>
  <div
    class="sc-chPdSV dwxkbg"
    data-testid="phrases-container"
  >
    <div
      class="sc-bxivhb eRFxNn"
    >
      <div
        class="sc-ifAKCX jJulbc"
      >
        <div
          class="sc-EHOje kGzamQ"
          data-testid="front"
        >
          <div
            class="sc-htoDjs sc-dnqmqq bhPwuV"
          >
            <div
              class="sc-iwsKbI dKyiiK"
              data-testid="name"
            >
              Test phrase
            </div>
          </div>
        </div>
        <div
          class="sc-bZQynM eprqWe"
          data-testid="back"
        >
          <div
            class="sc-htoDjs sc-gZMcBi bSzNzK"
          >
            <div
              class="sc-gqjmRU eJtqmJ"
              data-testid="meaning-0"
            >
              Meaning
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-bxivhb eRFxNn"
    >
      <div
        class="sc-ifAKCX jJulbc"
      >
        <div
          class="sc-EHOje kGzamQ"
          data-testid="front"
        >
          <div
            class="sc-htoDjs sc-dnqmqq bhPwuV"
          >
            <div
              class="sc-iwsKbI dKyiiK"
              data-testid="name"
            >
              Test phrase 2
            </div>
          </div>
        </div>
        <div
          class="sc-bZQynM eprqWe"
          data-testid="back"
        >
          <div
            class="sc-htoDjs sc-gZMcBi bSzNzK"
          >
            <div
              class="sc-gqjmRU bPjfci"
              data-testid="meaning-0"
            >
              Meaning 1
            </div>
            <div
              class="sc-gqjmRU bPjfci"
              data-testid="meaning-1"
            >
              Meaning 2
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
